File: python/sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def dfs(left, node):
            if not node: return 0
            if node.left == None and node.right == None:
                if left: return node.val
                else: return 0
            
            return dfs(True, node.left) + dfs(False, node.right)
        
        return dfs(False, root)

File: go/sum_of_left_leaves.go
package leetcode

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumOfLeftLeaves(root *TreeNode) int {
	return dfs(false, root)
}

func dfs(left bool, node *TreeNode) int {
	if node == nil {
		return 0
	}
	if node.Left == nil && node.Right == nil {
		if left {
			return node.Val
		} else {
			return 0
		}
	}
	return dfs(true, node.Left) + dfs(false, node.Right)
}
